application/app/Http/Resources/Sync/InstitutionUserSyncResource.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace App\Http\Resources\Sync;
+
+use App\Enums\InstitutionUserStatus;
+use App\Http\Resources\DepartmentResource;
+use App\Http\Resources\UserResource;
+use App\Models\InstitutionUser;
+use Illuminate\Http\Request;
+use Illuminate\Http\Resources\Json\JsonResource;
+use Illuminate\Support\Carbon;
+
+/**
+ * @mixin InstitutionUser
+ */
+class InstitutionUserSyncResource extends JsonResource
+{
+    /**
+     * @return array{
+     *     id: string,
+     *     user: UserResource,
+     *     institution_id: string,
+     *     department: DepartmentResource,
+     *     phone: string,
+     *     email: string,
+     *     status: InstitutionUserStatus,
+     *     updated_at: Carbon,
+     *     created_at: Carbon,
+     *     deleted_at: Carbon
+     * }
+     */
+    public function toArray(Request $request): array
+    {
+        return [
+            ...$this->only(
+                'id',
+                'email',
+                'phone',
+                'status',
+                'created_at',
+                'updated_at',
+                'deleted_at'
+            ),
+            'user' => new UserResource($this->user),
+            'department' => new DepartmentResource($this->department),
+            'institution_id' => $this->institution_id,
+        ];
+    }
+}
```

Comment:
> How I see it: it’s hard to predict what exact data other services will require, so it makes sense to me to just include ALL data about the entity (e.g. return $this->toArray() or something similiar).
> There will never be a need to "hide" sensitive data in these sync resources because they never leave the boundary of our back-end. They’re just passed from service to service, never to front-end (unless something goes horribly wrong).

I can think of a case where a developer who is not familiar with other service's logic can see the data from sync endpoint and may use it in way that it is not intended. For example if there are some fields that the service itself requires to work correctly and are only meant for internal usage (e.g. when using some polymorphic tables which will contain model references etc...)

> Returning all data would relieve us from having to update these resources when we add new columns (e.g. deactivation_date, archived_at

Somewhat we need to do it anyway when some new field needs to be shown in API endpoints for FE. It wouldn't be a big work to add one line to some sync resource too

Reply:
Okay, that makes sense!